src/generator.rs

```diff
@@ -228,14 +249,14 @@ Options={options}
 /// Path escaping as described in systemd.unit(5)
 ///
 /// `/./` components stripped away when parsing `mount-point =`
-fn mount_unit_name(path: &Path) -> String {
+fn mount_unit_name(path: &Path, suffix: &str) -> String {
```

Comment:
I think this should be renamed to `unit_name_from_path()` now.

As discussed above, this also needs a following fixup:
```diff
-                b'0'..=b'9' | b'a'..=b'z' | b'A'..=b'Z' | b'_' | b'.' => obuf.push(b),
+                b'0'..=b'9' | b'a'..=b'z' | b'A'..=b'Z' | b'_' | b'.' | b':' => obuf.push(b),
```
and please test that `"/dev/disk/by-path/virtio-pci-0000:08:00.0"` expands to `"dev-disk-by\\x2dpath-virtio\\x2dpci\\x2d0000:08:00.0.device"`.

Reply:
(separate commit please, for clarity)

Reply:
Fixed